src/classes/ADDR_Addresses_TEST.cls

```diff
@@ -998,6 +998,254 @@ public with sharing class ADDR_Addresses_TEST {
+    @description
+        Update Address' Mailing State (with State & Country Picklists enabled) when the State is empty
+        and/or mismatched.
+    verify:
+        New Address' State field is updated with the source sObject's StateCode value
+    **********************************************************************************************************/
+    static testMethod void testUpdateEmptyOrMismatchedAddressMailingStateWithAccountOrContactStateCode() {
+        if (!ADDR_Addresses_TDTM.isStateCountryPicklistsEnabled) {
+            return;
+        }
+
+        if (strTestOnly != '*' && strTestOnly != 'testUpdateEmptyOrMismatchedAddressMailingStateWithAccountOrContactStateCode') {
+            return;
+        }
+
+        Integer hhSize = 1;
+        Integer contactSize = 1;
+
+        // This creates a default Address for each HH
+        createHHTestData(hhSize, contactSize);
+
+        Contact con = listConT[0];
+        con.put('MailingStateCode', 'CA');
+        con.is_Address_Override__c = true;
+
+        Address__c addressWithoutState = listAddrT[0];
+        System.assertEquals('Washington', listAddrT[0].MailingState__c);
+
+        // Set the Mailing State to null
+        addressWithoutState.MailingState__c = null;
+        update addressWithoutState;
+        System.assertEquals(null, addressWithoutState.MailingState__c);
+
+        Test.startTest();
+        ADDR_Addresses_TDTM.updateEmptyOrMismatchedAddressMailingStateWithAccountOrContactStateCode(con, 'Mailing', addressWithoutState);
+        Test.stopTest();
+
+        System.assertEquals('CA', addressWithoutState.MailingState__c);
+    }
+
+
+    /*********************************************************************************************************
+    @description
+        Update Address' Mailing State (with State & Country Picklists enabled) when the State is empty
+        and/or mismatched.
+    verify:
+        New Address' State field is updated with the source sObject's StateCode value
+        New Address' State field (used as a Map's key) matches the source sObject's StateCode
+    **********************************************************************************************************/
+    static testMethod void testIsMatchingStateCode() {
+        if (!ADDR_Addresses_TDTM.isStateCountryPicklistsEnabled) {
+            return;
+        }
+
+        if (strTestOnly != '*' && strTestOnly != 'testIsMatchingStateCode') {
+            return;
+        }
+
+        Integer hhSize = 1;
+        Integer contactSize = 1;
+
+        // This creates a default Address for each HH
+        createHHTestData(hhSize, contactSize);
+
+        Contact con = listConT[0];
+        con.put('MailingStateCode', 'CA');
+        con.is_Address_Override__c = true;
+
+        Address__c addressWithState = listAddrT[0];
+        System.assertEquals('Washington', listAddrT[0].MailingState__c);
+
+        addressWithState.MailingState__c = null;
```

Comment:
Should we assert `ADDR_Addresses_TDTM.isMatchingStateCode` for an empty MailingState?